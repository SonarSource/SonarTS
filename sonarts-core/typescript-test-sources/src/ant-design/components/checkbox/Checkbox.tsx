import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import RcCheckbox from 'rc-checkbox';
import shallowEqual from 'shallowequal';
import CheckboxGroup from './Group';

export interface AbstractCheckboxProps {
  prefixCls?: string;
  className?: string;
  defaultChecked?: boolean;
  checked?: boolean;
  style?: React.CSSProperties;
  disabled?: boolean;
  onChange?: React.FormEventHandler<any>;
  onMouseEnter?: React.MouseEventHandler<any>;
  onMouseLeave?: React.MouseEventHandler<any>;
  value?: any;
}

export interface CheckboxProps extends AbstractCheckboxProps {
  indeterminate?: boolean;
}

export default class Checkbox extends React.Component<CheckboxProps, any> {
  static Group: typeof CheckboxGroup;
  static defaultProps = {
    prefixCls: 'ant-checkbox',
    indeterminate: false,
  };

  static contextTypes = {
    checkboxGroup: PropTypes.any,
  };

  shouldComponentUpdate(nextProps, nextState, nextContext) {
    return !shallowEqual(this.props, nextProps) ||
           !shallowEqual(this.state, nextState) ||
           !shallowEqual(this.context.checkboxGroup, nextContext.checkboxGroup);
  }

  render() {
    const { props, context } = this;
    const {
      prefixCls,
      className,
      children,
      indeterminate,
      style,
      onMouseEnter,
      onMouseLeave,
      ...restProps,
    } = props;
    const { checkboxGroup } = context;
    let checkboxProps: CheckboxProps = { ...restProps };
    if (checkboxGroup) {
      checkboxProps.onChange = () => checkboxGroup.toggleOption({ label: children, value: props.value });
      checkboxProps.checked = checkboxGroup.value.indexOf(props.value) !== -1;
      checkboxProps.disabled = 'disabled' in props ? props.disabled : checkboxGroup.disabled;
    }
    const classString = classNames(className, {
      [`${prefixCls}-wrapper`]: true,
    });
    const checkboxClass = classNames({
      [`${prefixCls}-indeterminate`]: indeterminate,
    });
    return (
      <label
        className={classString}
        style={style}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
        <RcCheckbox
          {...checkboxProps}
          prefixCls={prefixCls}
          className={checkboxClass}
        />
        {children !== undefined ? <span>{children}</span> : null}
      </label>
    );
  }
}
