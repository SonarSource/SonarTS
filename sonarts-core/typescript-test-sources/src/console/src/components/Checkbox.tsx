import * as React from 'react'
import {Icon, $v} from 'graphcool-styles'
import * as cn from 'classnames'

interface Props {
  checked: boolean
  onToggle: () => void
  children?: any
}

export default function Checkbox({checked, onToggle, children}: Props) {
  return (
    <div className='option-cell' onClick={onToggle}>
      <style jsx>{`
        .option-cell {
          @p: .inlineFlex, .itemsCenter, .pointer;
        }
        .option {
          @p: .flex, .itemsCenter, .justifyCenter, .br100, .ba, .bBlack20;
          width: 25px;
          height: 25px;
        }
        .option-cell:hover .option:not(.checked) {
          @p: .bgGreen10;
        }
        .option.checked {
          @p: .bn, .bgGreen20;
        }
      `}</style>
      <div className={cn('option', {checked})}>
        {checked && (
          <Icon
            src={require('graphcool-styles/icons/fill/check.svg')}
            color={checked ? $v.green : 'white'}
            width={17}
            height={17}
          />
        )}
      </div>
      {children || null}
    </div>
  )
}
