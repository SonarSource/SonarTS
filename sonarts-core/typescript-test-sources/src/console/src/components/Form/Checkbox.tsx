import * as React            from 'react' // tslint:disable-line
import {
  $p,
  variables,
  Icon,
}                            from 'graphcool-styles'
import styled                from 'styled-components'
import * as cx               from 'classnames'

const ConditionButton = styled.div`
  &:not(.${$p.bgBlue}):hover {
    background-color: ${variables.gray10};
  }
`

const CheckIcon = styled(Icon)`
  background-color: #4990E2;
  display: inline-flex !important;
  vertical-align: middle;
  padding: 5px;
  margin-right: 15px;
  border-radius: 50%;
`

const CheckIconDisabled = styled.div`
  background-color: ${variables.white};
  display: inline-flex !important;
  vertical-align: middle;
  padding: 12px;
  margin-right: 15px;
  border-radius: 50%;
  border: 1px solid black;
  opacity: 0.16;
`

const NestedCheckboxVerticalLine = styled.span`
  position: absolute;
  top: -25px;
  left: 0;
  height: 46px;
  border-left-width: 1px;
  border-left-style: solid;
`
const NestedCheckboxHorizontalLine = styled.span`
  position: absolute;
  width: 15px;
  height: 1px;
  top: 20px;
  left: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
`

interface Props {
  label: string
  checked: boolean
  onClick?: () => void
  nested?: boolean
  forceHighlightVerticalLine?: boolean
}

export default (props: Props) => {
  const nestedLines = props.nested && ([
    <NestedCheckboxVerticalLine key={0} className={cx({
       [$p.bBlue]: props.checked || props.forceHighlightVerticalLine,
       [$p.bBlack20]: !props.checked,
    })}/>,
    <NestedCheckboxHorizontalLine key={1} className={cx({
       [$p.bBlue]: props.checked,
       [$p.bBlack20]: !props.checked,
    })}/>,
  ])

  const icon = props.checked
    ? <CheckIcon
        color='white'
        src={require('../../assets/icons/check.svg')}
      />
    : <CheckIconDisabled />

  return (
    <div
      className={cx($p.relative, $p.mv6, $p.pointer, {
        [$p.pl16]: props.nested,
        [$p.z2]: !props.nested,
      })}
      onClick={props.onClick}
     >
      { nestedLines }
      { icon }
      <span className={cx($p.dib, $p.vMid, {
        [$p.black30]: !props.checked,
      })}>
        { props.label }
      </span>
    </div>
  )
}
