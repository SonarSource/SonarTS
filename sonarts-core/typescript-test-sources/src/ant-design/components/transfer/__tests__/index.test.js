/* eslint no-use-before-define: "off" */
import React from 'react';
import { render, mount } from 'enzyme';
import Transfer from '..';
import TransferList from '../list';
import TransferOperation from '../operation';
import TransferSearch from '../search';
import TransferItem from '../item';
import Button from '../../button';
import Checkbox from '../../checkbox';

const listCommonProps = {
  dataSource: [{
    key: 'a',
    title: 'a',
  }, {
    key: 'b',
    title: 'b',
  }, {
    key: 'c',
    title: 'c',
    disabled: true,
  }],
  selectedKeys: ['a'],
  targetKeys: ['b'],
  lazy: false,
};

const listDisabledProps = {
  dataSource: [{
    key: 'a',
    title: 'a',
    disabled: true,
  }, {
    key: 'b',
    title: 'b',
  }],
  selectedKeys: ['a', 'b'],
  targetKeys: [],
  lazy: false,
};

const searchTransferProps = {
  dataSource: [
    {
      key: '0',
      title: 'content1',
      description: 'description of content1',
      chosen: false,
    },
    {
      key: '1',
      title: 'content2',
      description: 'description of content2',
      chosen: false,
    },
    {
      key: '2',
      title: 'content3',
      description: 'description of content3',
      chosen: false,
    },
    {
      key: '3',
      title: 'content4',
      description: 'description of content4',
      chosen: false,
    },
    {
      key: '4',
      title: 'content5',
      description: 'description of content5',
      chosen: false,
    },
    {
      key: '5',
      title: 'content6',
      description: 'description of content6',
      chosen: false,
    },
  ],
  selectedKeys: [],
  targetKeys: ['3', '4'],
  lazy: false,
};

describe('Transfer', () => {
  it('should render correctly', () => {
    const wrapper = render(<Transfer {...listCommonProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should move selected keys to corresponding list', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onChange={handleChange} />);
    wrapper.find(TransferOperation).find(Button).at(1).simulate('click'); // move selected keys to right list
    expect(handleChange).toHaveBeenCalledWith(['a', 'b'], 'right', ['a']);
  });

  it('should move selected keys expect disabled to corresponding list', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<Transfer {...listDisabledProps} onChange={handleChange} />);
    wrapper.find(TransferOperation).find(Button).at(1).simulate('click'); // move selected keys to right list
    expect(handleChange).toHaveBeenCalledWith(['b'], 'right', ['b']);
  });

  it('should uncheck checkbox when click on checked item', () => {
    const handleSelectChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onSelectChange={handleSelectChange} />);
    wrapper.find(TransferItem).filterWhere(n => n.prop('item').key === 'a').simulate('click');
    expect(handleSelectChange).toHaveBeenLastCalledWith([], []);
  });

  it('should check checkbox when click on unchecked item', () => {
    const handleSelectChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onSelectChange={handleSelectChange} />);
    wrapper.find(TransferItem).filterWhere(n => n.prop('item').key === 'b').simulate('click');
    expect(handleSelectChange).toHaveBeenLastCalledWith(['a'], ['b']);
  });

  it('should not check checkbox when click on disabled item', () => {
    const handleSelectChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onSelectChange={handleSelectChange} />);
    wrapper.find(TransferItem).filterWhere(n => n.prop('item').key === 'c').simulate('click');
    expect(handleSelectChange).not.toHaveBeenCalled();
  });

  it('should check all item when click on check all', () => {
    const handleSelectChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onSelectChange={handleSelectChange} />);
    wrapper.find('.ant-transfer-list-header input[type="checkbox"]')
      .filterWhere(n => !n.prop('checked')).simulate('change');
    expect(handleSelectChange).toHaveBeenCalledWith(['a'], ['b']);
  });

  it('should uncheck all item when click on uncheck all', () => {
    const handleSelectChange = jest.fn();
    const wrapper = mount(<Transfer {...listCommonProps} onSelectChange={handleSelectChange} />);
    wrapper.find('.ant-transfer-list-header input[type="checkbox"]')
      .filterWhere(n => n.prop('checked')).simulate('change');
    expect(handleSelectChange).toHaveBeenCalledWith([], []);
  });

  it('should call `filterOption` when use input in search box', () => {
    const filterOption = (inputValue, option) => inputValue === option.title;
    const wrapper = mount(<Transfer {...listCommonProps} showSearch filterOption={filterOption} />);
    wrapper.find(TransferSearch).at(0).find('input').simulate('change', { target: { value: 'a' } });
    expect(wrapper.find(TransferList).at(0).find(TransferItem).find(Checkbox)).toHaveLength(1);
  });

  it('should display the correct count of items when filter by input', () => {
    const filterOption = (inputValue, option) => option.description.indexOf(inputValue) > -1;
    const renderFunc = item => item.title;
    const wrapper = mount(
      <Transfer
        {...searchTransferProps}
        showSearch
        filterOption={filterOption}
        render={renderFunc}
      />
    );
    wrapper.find(TransferSearch).at(0).find('input').simulate('change', { target: { value: 'content2' } });
    expect(wrapper.find(TransferList).at(0).find('.ant-transfer-list-header-selected > span').at(0)
      .first()
      .text()
      .trim()).toEqual('1');
  });

  it('should just check the filtered item when click on check all after search by input', () => {
    const filterOption = (inputValue, option) => option.description.indexOf(inputValue) > -1;
    const renderFunc = item => item.title;
    const handleSelectChange = jest.fn();
    const wrapper = mount(
      <Transfer
        {...searchTransferProps}
        showSearch
        filterOption={filterOption}
        render={renderFunc}
        onSelectChange={handleSelectChange}
      />
    );
    wrapper.find(TransferSearch).at(0).find('input')
      .simulate('change', { target: { value: 'content2' } });
    wrapper.find(TransferList).at(0).find('.ant-transfer-list-header input[type="checkbox"]').filterWhere(n => !n.prop('checked'))
      .simulate('change');
    expect(handleSelectChange).toHaveBeenCalledWith(['1'], []);
  });

  it('should transfer just the filtered item after search by input', () => {
    const filterOption = (inputValue, option) => option.description.indexOf(inputValue) > -1;
    const renderFunc = item => item.title;
    const handleChange = jest.fn();
    const handleSelectChange = (sourceSelectedKeys, targetSelectedKeys) => {
      wrapper.setProps({
        selectedKeys: [...sourceSelectedKeys, ...targetSelectedKeys],
      });
    };
    const wrapper = mount(
      <Transfer
        {...searchTransferProps}
        showSearch
        filterOption={filterOption}
        render={renderFunc}
        onSelectChange={handleSelectChange}
        onChange={handleChange}
      />
    );
    wrapper.find(TransferSearch).at(0).find('input')
      .simulate('change', { target: { value: 'content2' } });
    wrapper.find(TransferList).at(0).find('.ant-transfer-list-header input[type="checkbox"]').filterWhere(n => !n.prop('checked'))
      .simulate('change');
    wrapper.find(TransferOperation).find(Button).at(1).simulate('click');
    expect(handleChange).toHaveBeenCalledWith(['1', '3', '4'], 'right', ['1']);
  });

  it('should check correctly when there is a search text', () => {
    const props = { ...listCommonProps };
    delete props.targetKeys;
    delete props.selectedKeys;
    const handleSelectChange = jest.fn();
    const wrapper = mount(
      <Transfer {...props} showSearch onSelectChange={handleSelectChange} render={item => item.title} />
    );
    wrapper.find(TransferItem).filterWhere(n => n.prop('item').key === 'b').simulate('click');
    expect(handleSelectChange).toHaveBeenLastCalledWith(['b'], []);
    wrapper.find(TransferSearch).at(0).find('input').simulate('change', { target: { value: 'a' } });
    wrapper.find(TransferList).at(0).find('.ant-transfer-list-header input[type="checkbox"]').simulate('change');
    expect(handleSelectChange).toHaveBeenLastCalledWith(['b', 'a'], []);
    wrapper.find(TransferList).at(0).find('.ant-transfer-list-header input[type="checkbox"]').simulate('change');
    expect(handleSelectChange).toHaveBeenLastCalledWith(['b'], []);
  });

  it('should show sorted targetkey', () => {
    const sortedTargetKeyProps = {
      dataSource: [{
        key: 'a',
        title: 'a',
      }, {
        key: 'b',
        title: 'b',
      }, {
        key: 'c',
        title: 'c',
      }],
      targetKeys: ['c', 'b'],
      lazy: false,
    };
    const wrapper = render(<Transfer {...sortedTargetKeyProps} render={item => item.title} />);
    expect(wrapper).toMatchSnapshot();
  });
});
