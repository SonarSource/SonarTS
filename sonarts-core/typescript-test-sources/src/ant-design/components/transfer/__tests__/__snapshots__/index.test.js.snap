// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer should render correctly 1`] = `
<div
  class="ant-transfer"
>
  <div
    class="ant-transfer-list"
  >
    <div
      class="ant-transfer-list-header"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox ant-checkbox-checked"
        >
          <input
            checked=""
            class="ant-checkbox-input"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span
        class="ant-transfer-list-header-selected"
      >
        <span>
          1/2 
        </span>
        <span
          class="ant-transfer-list-header-title"
        />
      </span>
    </div>
    <div
      class="ant-transfer-list-body"
    >
      <ul
        class="ant-transfer-list-content"
      >
        <li
          class="ant-transfer-list-content-item"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                checked=""
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span />
        </li>
        <li
          class="ant-transfer-list-content-item ant-transfer-list-content-item-disabled"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox ant-checkbox-disabled"
            >
              <input
                class="ant-checkbox-input"
                disabled=""
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span />
        </li>
      </ul>
      <div
        class="ant-transfer-list-body-not-found"
      >
        Not Found
      </div>
    </div>
  </div>
  <div
    class="ant-transfer-operation"
  >
    <button
      class="ant-btn ant-btn-primary ant-btn-sm"
      disabled=""
      type="button"
    >
      <span>
        <i
          class="anticon anticon-left"
        />
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-sm"
      type="button"
    >
      <span>
        <i
          class="anticon anticon-right"
        />
      </span>
    </button>
  </div>
  <div
    class="ant-transfer-list"
  >
    <div
      class="ant-transfer-list-header"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span
        class="ant-transfer-list-header-selected"
      >
        <span>
          1 
        </span>
        <span
          class="ant-transfer-list-header-title"
        />
      </span>
    </div>
    <div
      class="ant-transfer-list-body"
    >
      <ul
        class="ant-transfer-list-content"
      >
        <li
          class="ant-transfer-list-content-item"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span />
        </li>
      </ul>
      <div
        class="ant-transfer-list-body-not-found"
      >
        Not Found
      </div>
    </div>
  </div>
</div>
`;

exports[`Transfer should show sorted targetkey 1`] = `
<div
  class="ant-transfer"
>
  <div
    class="ant-transfer-list"
  >
    <div
      class="ant-transfer-list-header"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span
        class="ant-transfer-list-header-selected"
      >
        <span>
          1 
        </span>
        <span
          class="ant-transfer-list-header-title"
        />
      </span>
    </div>
    <div
      class="ant-transfer-list-body"
    >
      <ul
        class="ant-transfer-list-content"
      >
        <li
          class="ant-transfer-list-content-item"
          title="a"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span>
            a
          </span>
        </li>
      </ul>
      <div
        class="ant-transfer-list-body-not-found"
      >
        Not Found
      </div>
    </div>
  </div>
  <div
    class="ant-transfer-operation"
  >
    <button
      class="ant-btn ant-btn-primary ant-btn-sm"
      disabled=""
      type="button"
    >
      <span>
        <i
          class="anticon anticon-left"
        />
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-sm"
      disabled=""
      type="button"
    >
      <span>
        <i
          class="anticon anticon-right"
        />
      </span>
    </button>
  </div>
  <div
    class="ant-transfer-list"
  >
    <div
      class="ant-transfer-list-header"
    >
      <label
        class="ant-checkbox-wrapper"
      >
        <span
          class="ant-checkbox"
        >
          <input
            class="ant-checkbox-input"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span
        class="ant-transfer-list-header-selected"
      >
        <span>
          2 
        </span>
        <span
          class="ant-transfer-list-header-title"
        />
      </span>
    </div>
    <div
      class="ant-transfer-list-body"
    >
      <ul
        class="ant-transfer-list-content"
      >
        <li
          class="ant-transfer-list-content-item"
          title="c"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span>
            c
          </span>
        </li>
        <li
          class="ant-transfer-list-content-item"
          title="b"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
          <span>
            b
          </span>
        </li>
      </ul>
      <div
        class="ant-transfer-list-body-not-found"
      >
        Not Found
      </div>
    </div>
  </div>
</div>
`;
